assets/src/edit-story/app/media/utils/getTypeFromMime.js

```diff
@@ -18,16 +18,15 @@
  * Infer element type from mime type of its resource
  *
  * @param {string} mimeType Mime type.
- * @return {string} Element type.
+ * @return {string|null} Element type.
  */
 const getTypeFromMime = (mimeType) => {
```

Comment:
I personally have never liked this function. I think it is very limited. A mime type is `image/gif`. Why not split and shift. 
```
const getTypeFromMime = (mimeType) => mimeType.split('/').shift();
```

Reply:
Yep, seems reasonable, I guess this function is just for returning type, not type from our supported subset? @obetomuniz

Reply:
Yup. That’s correct @merapi. It’s for normalization. PDF for instance will be `application/pdf`, returning "application" value on the split/shift method. Is it expected across the editor? What if I send a JSON file, which have the mimetype "application/json"?

Reply:
Right, element type `application` doesn't make sense for PDF and JSON (although, not sure if we have any use case). For our current use-case - both will do the job.

Reply:
The type should be correct. Returning null doesn't make a lot of sense.

Reply:
What is the definition of _correct_ element type here? `application` for PDF (or anything else) doesn't sound like one.

Reply:
PDF is a weird example, but a type is a wide group of what the mime type is. So `image/gif` is an image, `video/mp4` is a video, `text/html` is text and `audio/mp3` is audio. So I thin the logic I provided should for all those use cases. What you do with the type is a nothing matter.

Reply:
Just saying: for now - it does the job.
If more cases will arrive, like handling `text/html` then `application/pdf` is also likely, for example, you want to attach some kind of document (text or pdf) - we will have to change both of these functions. No such thing for now.

Reply:
That works for me, but it should be documented. Ideally with some tests & inline docs.

At the very least, we should have a test verifying that the media library renders fine when there are PDFs (if possible).